packages/client/src/components/BlockComponent.svelte

```diff
@@ -23,6 +23,8 @@
   // Create a fake component instance so that we can use the core Component
   // to render this part of the block, taking advantage of binding enrichment
   $: id = `${block.id}-${context ?? rand}`
+  $: parentId = $component?.id
+  $: inBuilder = $builderStore.inBuilder
```

Comment:
Just some dereferencing to primitives, to allow svelte to trigger the reactive registration below as infrequently as possible.